README.md

```diff
@@ -352,18 +354,37 @@ buildctl build ... --import-cache type=local,src=path/to/input-dir
 
 The directory layout conforms to OCI Image Spec v1.0.
 
+#### GitHub Actions cache (experimental)
```

Comment:
I might be missing it, but something I've struggled with in the past is understanding how the various caches actually work. This one, from the bash example, seems to, maybe (?) push the cache to docker.io, I guess? Or maybe it uses Github's build in cache? I'm not sure. Maybe a sentence or two describing how (and possibly where?) it works would be helpful? Even if it just said, "This cache is (will be) the preferred and most performant cache for Github Actions."

It's hard from the outside understanding all the different cache approaches. If this is documented somewhere else, maybe a pointer there would help?

Reply:
@crazy-max ^

Reply:
@mlissner 

> This one, from the bash example, seems to, maybe (?) push the cache to docker.io

GitHub Actions cache exporter backend uses the [GitHub Cache API](https://github.com/tonistiigi/go-actions-cache/blob/master/api.md) to fetch and upload cache blobs. That's why this type of cache should be used exclusively in a GitHub Action workflow as the `url` (`$ACTIONS_CACHE_URL`) and `token` (`$ACTIONS_RUNTIME_TOKEN`) are populated when a [Runner (workflow)](https://github.com/actions/runner) is started. To recap, it is unlikely that you can use this cache locally because these two variables are ephemeral and associated with your GitHub repository (pretty much like the GITHUB_TOKEN secret).

> Maybe a sentence or two describing how (and possibly where?) it works would be helpful?

I can enhance this PR with the explanation below if it looks good to you?

> It's hard from the outside understanding all the different cache approaches. If this is documented somewhere else, maybe a pointer there would help?

Each one of them is explained in [the Export cache section](https://github.com/moby/buildkit#export-cache) of the README but maybe we could split/move [--export-cache](https://github.com/moby/buildkit#--export-cache-options) and [--import-cache](https://github.com/moby/buildkit#--import-cache-options) sections to avoid confusion around the different available options.

Reply:
From my perspective, I came to the build action not even knowing what buildkit is. Maybe for docs an explanation of how the github cache works would be good, with a link to the export cache docs in buildkit? I honestly wouldn't have thought to look there for this information, so a link there would be helpful.

Reply:
I guess my point is that people (like me) might not know much when they come to the Github Action, so some more docs or pointers to docs would be helpful. I'm not sure which solution is best though.

Reply:
> Maybe for docs an explanation of how the github cache works would be good, with a link to the export cache docs in buildkit? I honestly wouldn't have thought to look there for this information, so a link there would be helpful.

Sure we can enhance documentation in https://github.com/docker/build-push-action/pull/406 with the explanations I gave below. Let's talk about it there.